src/main/java/com/mineboundteam/minebound/magic/PassiveSpellItem.java

```diff
@@ -8,20 +8,22 @@
 import net.minecraft.core.NonNullList;
 import net.minecraft.world.entity.EquipmentSlot;
 import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
 
 public abstract class PassiveSpellItem extends SpellItem {
     public PassiveSpellItem(Properties properties, ArmorTier level) {
         super(properties, level);
     }
 
-    protected <T extends PassiveSpellItem> List<ItemStack> getEquippedSpellsOfType(Class<T> type, Player player) {
-        NonNullList<ItemStack> spells = NonNullList.create();
+    @SuppressWarnings("unchecked")
+    protected static <T extends PassiveSpellItem> List<T> getEquippedSpellsOfType(Class<T> type, Player player) {
+        NonNullList<T> spells = NonNullList.create();
         for (EquipmentSlot e : EquipmentSlot.values())
             player.getItemBySlot(e).getCapability(ArmorSpellsProvider.ARMOR_PASSIVE_SPELLS).ifPresent(slots -> {
                 for (ItemStack item : slots.items) {
-                    if (type.isInstance(item))
-                        spells.add(item);
+                    if (type.isInstance(item.getItem()))
+                        spells.add((T) item.getItem());
```

Comment:
This is less of a change request and more of a comment:  
If you cast it directly to the item here, it will grab the registry's instance of the item under that name, not the itemstack. In this case, it doesn't really matter, but it will ignore any nbt metadata that the items in the armor might have stored, if that's something we ever care to think about.   
It's not something that necessarily needs changed, but it is something to consider. ItemStack stores player's item instance, Item stores the registry's

Reply:
I think if we end up in a situation where we want that, we can make a second function that does this. Doing the cast inside this function significantly cleans up the code outside it and makes it easier to read.